### ISSUE TITLE
Table headers and column contents

### ISSUE THREAD

**Issue by kfischer** [jajm/koha-staff-interface-redesign#22]
When the mock-ups were published a lot of people commented on needing the lines between columns for orientation. 

What definitely makes orientation harder is the different alignment of table headers and contents without the lines. You can see on the screenshot below that content and headers are not aligned well in a lot of cases, standing out is 'date accessioned':

![grafik](https://user-images.githubusercontent.com/425526/189335805-8647f3d2-2317-4d16-b514-49ab116b8404.png)

**Comment by kfischer** [comment]
For me this works well/better than before.  With the lines, the different alignment of headers and content doesn't create as much of a visual mismatch. 

Adding 2 screenshot of the current look for people to see as this will be different than mock-ups (also lost the even-uneven-coloring I think?)

![grafik](https://user-images.githubusercontent.com/425526/190139579-02cb010a-7009-4115-b034-1dbccaaa3d86.png)

![grafik](https://user-images.githubusercontent.com/425526/190139684-5cbeb6db-d58b-446a-8e23-fe178b364724.png)

**Comment by kfischer** [comment]
I think it would be good to get more opinions here before closing.

Maybe an alternative:
- Add "striping" of the lines using background color
- Text-align the table headers left

**Comment by kfischer** [comment]
Ok, after talking to @mrenvoize: There is currently striping, but maybe I got a bad screen, because I am not seeing it... maybe it should be less subtle in coloring :)

**Comment by mrenvoize** [comment]
The contrast between even and odd rows is tiny: [1.05:1](https://webaim.org/resources/contrastchecker/?fcolor=F9F9F9&bcolor=FFFFFF).

It's using the same colours as in master, but I do agree it's so small that I imagine it's completely lost on a lot of monitors or just lost by people's eyes.. we could certainly up it a bit somehow.

I'm not sure we need full cell borders personally.. I'd like to try it with just vertical borders to see how that looks.. I might have a bit of an experiment.

**Comment by mrenvoize** [comment]
A little experiment with striping.. 
* Set the background of the 'tab-content' to white (#fff) instead of transparent 
* Use the background-color we use on the 'body' as the new even row highlight color (#f3f4f4)

![screenshot-staff-gui sandboxes biblibre eu-2022 09 14-13_56_54](https://user-images.githubusercontent.com/557612/190160070-fbf10bf4-750f-4d30-8724-56b20e37497d.png)

Is that clearer to you @kfischer ?

I also had a little play with borders in that screenshot.. removing horizontal borders except for the header row and keeping only the 'border-left' for other fields (though I'd like to also drop that for the first column)

**Comment by mrenvoize** [comment]
Still only a [1.1:1](https://webaim.org/resources/contrastchecker/?fcolor=F3F4F4&bcolor=FFFFFF) contrast ratio, but to my eye's it's much clearer and it uses a color we've already picked so is consistent.

**Comment by kfischer** [comment]
> 

Yes, I can see that one on my screen :) I think if we are to remove the borders, which we can try, we need to have the striping stand out a bit more.

**Comment by mrenvoize** [comment]
![screenshot-staff-gui sandboxes biblibre eu-2022 09 14-14_21_37](https://user-images.githubusercontent.com/557612/190383934-3d545dd7-7efe-4e07-86c1-4ad69c695752.png)

**Comment by kfischer** [comment]
> ![screenshot-staff-gui sandboxes biblibre eu-2022 09 14-13_56_54](https://user-images.githubusercontent.com/557612/190165607-723a3db9-4bcb-44d1-99b9-b41237b25f70.png)I

This is better :)

**Comment by mrenvoize** [comment]
This is the overriden CSS I ended up with in my console: 

```css
.tab-content {
  background-color: #ffffff;
}

tbody tr:nth-child(odd):not(.dtrg-group):not(.active):not(.ok) td:not(.bg-danger):not(.bg-warning):not(.bg-info):not(.bg-success):not(.bg-primary) {
    background-color: #f3f4f4;
}

td, th {
    border-left: 1px solid #bcbcbc;
}

td:first-child, th:first-child {
    border-left: none;
}
```

Hope it helps as a clue :)

**Comment by jajm** [comment]
> Hope it helps as a clue :)

Very useful, thanks!

**Comment by kfischer** [comment]
I quite like the newest version :) 

* Can see the row-striping on my screen now, which makes a huge difference
* Like the removed lines between rows

Only wondering, because some tables show a line above the header row, while others (like items table on staff detail) don't:

![grafik](https://user-images.githubusercontent.com/425526/191024132-f831f98e-0faf-4209-9ba4-11b2bbc032f5.png)

**Comment by jajm** [comment]
I'll take a look at this

**Comment by mrenvoize** [comment]
This is starting to look really great.. sorry it's such hard work catching all cases @jajm, we really need to get more consistent as Devs with our markup.. then css changes like this would be much simpler.

**Comment by jajm** [comment]
Tables should have the same look now.
- 2 pixels border below `<thead>` and above `<tfoot>`
- No border above or below table cells (in thead, tbody and tfoot)
- No border around tables

 A few examples:

![image](https://user-images.githubusercontent.com/306836/192489964-6f17f5fe-12f2-440b-8d70-993f0dec0bb8.png)

![image](https://user-images.githubusercontent.com/306836/192490082-6cec2cac-1548-45fe-b13c-2c8f1ab04fd7.png)

![image](https://user-images.githubusercontent.com/306836/192490230-cd0b2605-d2af-4629-a608-7dbe0e7d70b8.png)

![image](https://user-images.githubusercontent.com/306836/192490506-8d18e198-0b2f-4048-bdf8-f9efa9d60440.png)

**Comment by mrenvoize** [comment]
Looks great to me now :)

**Comment by mrenvoize** [comment]
Closable.. I'm chasing any outstanding cases of a missing page-section (i.e. ensuring all tables are on a white background overall to prevent the odd highlighting getting lost against the overall page background) on other bugs.
